content/livesync/Setup

```diff
@@ -0,0 +1,190 @@
+</aside>
+
+"Sign up":https://ably.com/sign-up to Ably to create an API key in the "dashboard":https://ably.com/dashboard or use the "Control API":/account/control-api to create an API programmatically.
+
+API keys and tokens have a set of capabilities assigned to them that specify which operations, such as @subscribe@ or @publish@ can be performed on which resources. To use the Ably Realtime Data Models SDK, the API key requires the following "capabilities":/auth/capabilities#58: 
+
+* Publish
+* Subscribe
+* Presence
+* History
+
+h2(#install). Install
+
+The Realtime Data Models SDK is a JavaScript (TypeScript) library that enables you to create live and observable data models in your frontend application. These models remain synchronized with the realtime state of your database model. You can easily integrate this SDK into your project regardless of your frontend framework preference.
+
+Your backend publishes mutation events to Ably. The Realtime Data Models SDK updates your frontend app's local state. You can also pair the SDK with *Ably's Database Connector* to transmit transactional change events with your database mutations.
+
+The Realtime Data Models SDK is available as an "NPM module":#npm.
+
+h3(#npm1). Use NPM
+
+Install the Ably JavaScript SDK and the Realtime Data Models SDK:
+
+```
+npm install ably @ably-labs/models
+```
+
+
+h3(#npm-link). Use the NPM link
+
+Clone this repository and run @npm link@:
+
+```
+git clone git@github.com:ably-labs/models.git
+cd models
+npm link
+```
+
+From your project, link to the cloned project and build:
+
+```
+cd ./your/project
+npm link @ably-labs/models
+pushd ./node_modules/@ably-labs/models
+npm run build
+popd
+```
+
+You should now be able to import @@ably-labs/models@ in your project.
+
+h2(#instantiate). Instantiate
+
+To instantiate the Realtime Data Models SDK, create an "Ably client":/getting-started/setup?lang=javascript and pass it into the @ModelsClient@ constructor:
+
+```
+import ModelsClient from '@ably-labs/models';
+import { Realtime } from 'ably';
+
+const ably = new Realtime.Promise({ key: 'YOUR_ABLY_API_KEY' });
+const modelsClient = new ModelsClient({ ably });
+```
+
+h2(#create-model). Create a model
+
+A @model@ represents a live, observable data model supported by the database.
+
+To create a model, you need to:
+
+1. Define the model's data structure in the frontend application.
+2. Initialize the model.
+3. Update the model based on events from the backend.
+4. Determine how end-users can modify the model.
+
+h3(#structure). Ably Realtime Data Models structure
+
+The structure of the models data as it is used in the frontend application:
+
+```
+type Post = {
+  id: number;
+  text: string;
+  comments: string[];
+}
+```
+
+h3(#initialise). Initialise
+
+An example function used by the model to initialise with the correct data from your backend:
+
+```
+async function sync() {
```

Comment:
Do we need to indicate the return type here? We are currently not using the `Post` type defined above in these docs at all. In reality, the type of the model is inferred from the return type of the `sync` function.

The sync function also needs to return a sequence ID which isn't covered here.

See: https://github.com/ably-labs/models/blob/main/docs/usage.md#sync-function

Reply:
@mschristensen Mike will shorten the code snippets